vassal-app/src/main/java/bsh/BeanShellExpressionValidator.java

```diff
@@ -38,7 +40,54 @@
   protected String error;
   protected boolean valid;
 
-  protected final String SUPPORTED_STRING_METHODS = ".length,.contains,.startsWith,.endsWith,.matches,.indexOf,.lastIndexOf,.substring,.replace,";
+  /**
+   * List of all Java 15 String functions that can be applied to property names directly within Beanshell.
+   * (i.e. those with return values and argument types that Vassal supports).
+   *
+   * This list is used to recognize and evaluate constructs of the form
+   *
+   *    stringVariable.function()
+   *
+   * And allow stringVariable to be used directly within beanshell with Java syntax,
+   * rather than having to resort to
+   *
+   *    GetProperty("stringVariable").function()
+   *
+   */
+  private static final Set<String> supportedStringFunctions;
```

Comment:
You can add all these without a `static` block using `Set.of(...)`:
```
  private static final Set<String> supportedStringFunctions = Set.of( 
    ".compareTo",
    ".compareToIgnoreCase",
    ...
  );
```

Reply:
Updated